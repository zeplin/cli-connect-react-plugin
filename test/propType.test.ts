import Plugin from "../src";
import { logger } from "./helper/logger";

describe("Connected Components React Plugin - PropTypes", () => {
    test("Component.jsx snippet creation", async () => {
        const plugin = new Plugin();

        const componentCode = await plugin.process(
            {
                path: "test/samples/jsx-class/Component.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("ComponentWithChildren.jsx snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/jsx-class/ComponentWithChildren.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("ComponentWithProps.jsx snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/jsx-class/ComponentWithProps.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("ComponentWithChildrenAndProps.jsx snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/jsx-class/ComponentWithChildrenAndProps.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("ComponentWithMemoization.jsx snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/jsx-class/ComponentWithMemoization.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("MultiExportComponentWithProps.jsx snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/jsx-class/MultiExportComponentWithProps.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("ComponentWithChildrenAndProps.jsx snippet creation with single export resolver", async () => {
        const processor = new Plugin();
        await processor.init({
            components: [],
            logger,
            config: {
                reactDocgenResolver: "findExportedComponentDefinition"
            }
        });

        const componentCode = await processor.process(
            {
                path: "test/samples/jsx-class/ComponentWithChildrenAndProps.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });
});
