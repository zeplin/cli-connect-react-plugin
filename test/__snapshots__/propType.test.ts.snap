// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Components React Plugin - PropTypes Component.jsx snippet creation 1`] = `
Object {
  "description": "Component description.",
  "lang": "jsx",
  "snippet": "<MyComponent />",
}
`;

exports[`Connected Components React Plugin - PropTypes ComponentWithChildren.jsx snippet creation 1`] = `
Object {
  "description": "Component description.",
  "lang": "jsx",
  "snippet": "<MyComponent>
  {children}
</MyComponent>",
}
`;

exports[`Connected Components React Plugin - PropTypes ComponentWithChildrenAndProps.jsx snippet creation 1`] = `
Object {
  "description": "Component description.",
  "lang": "jsx",
  "snippet": "<MyComponent
  optionalArray={array}
  optionalBool={bool}
  optionalFunc={func}
  optionalNumber={number}
  optionalObject={object}
  optionalString={string}
  optionalSymbol={symbol}
  optionalNode={node}
  optionalElement={element}
  optionalElementType={elementType}
  optionalFoo={instanceOf(Foo)}
  optionalEnum={enum}
  optionalUnion={union[string|number|instanceOf(Foo)]}
  optionalArrayOf={arrayOf[number]}
  optionalObjectOf={objectOf[number]}
  optionalObjectWithShape={shape}
  optionalObjectWithStrictShape={exact}
  requiredFunc={func}
  requiredAny={any}
  customProp={() => {}}
  customArrayProp={arrayOf[custom]}
  customObjectOfProp={objectOf[custom]}>
  {children}
</MyComponent>",
}
`;

exports[`Connected Components React Plugin - PropTypes ComponentWithChildrenAndProps.jsx snippet creation with single export resolver 1`] = `
Object {
  "description": "Component description.",
  "lang": "jsx",
  "snippet": "<MyComponent
  optionalArray={array}
  optionalBool={bool}
  optionalFunc={func}
  optionalNumber={number}
  optionalObject={object}
  optionalString={string}
  optionalSymbol={symbol}
  optionalNode={node}
  optionalElement={element}
  optionalElementType={elementType}
  optionalFoo={instanceOf(Foo)}
  optionalEnum={enum}
  optionalUnion={union[string|number|instanceOf(Foo)]}
  optionalArrayOf={arrayOf[number]}
  optionalObjectOf={objectOf[number]}
  optionalObjectWithShape={shape}
  optionalObjectWithStrictShape={exact}
  requiredFunc={func}
  requiredAny={any}
  customProp={() => {}}
  customArrayProp={arrayOf[custom]}
  customObjectOfProp={objectOf[custom]}>
  {children}
</MyComponent>",
}
`;

exports[`Connected Components React Plugin - PropTypes ComponentWithMemoization.jsx snippet creation 1`] = `
Object {
  "description": "Component description.",
  "lang": "jsx",
  "snippet": "<MyComponent
  optionalArray={array}
  optionalBool={bool}
  optionalFunc={func}
  optionalNumber={number}
  optionalObject={object}
  optionalString={string}
  optionalSymbol={symbol}
  optionalNode={node}
  optionalElement={element}
  optionalElementType={elementType}
  optionalFoo={instanceOf(Foo)}
  optionalEnum={enum}
  optionalUnion={union[string|number|instanceOf(Foo)]}
  optionalArrayOf={arrayOf[number]}
  optionalObjectOf={objectOf[number]}
  optionalObjectWithShape={shape}
  optionalObjectWithStrictShape={exact}
  requiredFunc={func}
  requiredAny={any}
  customProp={() => {}}
  customArrayProp={arrayOf[custom]}
  customObjectOfProp={objectOf[custom]}>
  {children}
</MyComponent>",
}
`;

exports[`Connected Components React Plugin - PropTypes ComponentWithProps.jsx snippet creation 1`] = `
Object {
  "description": "Component description.",
  "lang": "jsx",
  "snippet": "<MyComponent
  optionalArray={array}
  optionalBool={bool}
  optionalFunc={func}
  optionalNumber={number}
  optionalObject={object}
  optionalString={string}
  optionalSymbol={symbol}
  optionalNode={node}
  optionalElement={element}
  optionalElementType={elementType}
  optionalFoo={instanceOf(Foo)}
  optionalEnum={enum}
  optionalUnion={union[string|number|instanceOf(Foo)]}
  optionalArrayOf={arrayOf[number]}
  optionalObjectOf={objectOf[number]}
  optionalObjectWithShape={shape}
  optionalObjectWithStrictShape={exact}
  requiredFunc={func}
  requiredAny={any}
  customProp={() => {}}
  customArrayProp={arrayOf[custom]}
  customObjectOfProp={objectOf[custom]} />",
}
`;

exports[`Connected Components React Plugin - PropTypes MultiExportComponentWithProps.jsx snippet creation 1`] = `
Object {
  "description": "Component 1 description. Only this one should be shown",
  "lang": "jsx",
  "snippet": "<MyComponent1
  optionalArray={array}
  optionalBool={bool}
  optionalFunc={func}
  optionalNumber={number}
  optionalObject={object}
  optionalString={string}
  optionalSymbol={symbol}
  optionalNode={node}
  optionalElement={element}
  optionalElementType={elementType}
  optionalFoo={instanceOf(Foo)}
  optionalEnum={enum}
  optionalUnion={union[string|number|instanceOf(Foo)]}
  optionalArrayOf={arrayOf[number]}
  optionalObjectOf={objectOf[number]}
  optionalObjectWithShape={shape}
  optionalObjectWithStrictShape={exact}
  requiredFunc={func}
  requiredAny={any}
  customProp={() => {}}
  customArrayProp={arrayOf[custom]}
  customObjectOfProp={objectOf[custom]} />

<MyComponent2
  optionalArray={array}
  optionalBool={bool}
  optionalFunc={func}
  optionalNumber={number}
  optionalObject={object}
  optionalString={string}
  optionalSymbol={symbol} />",
}
`;
